File: CivitasRuby/Sorpresa_SalirCarcel.rb
module Civitas
    class Sorpresa_SalirCarcel < Sorpresa
        public_class_method :new
        
        def initialize(mazo)
            
            super('Queda libre de la carcel.')
            @mazo=mazo

        end

        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual,todos)
                informe(actual, todos)
                obtiene=true
                i=0
                while i<todos.size-1 && !obtiene
                
                    if(todos[i].tieneSalvoconducto)
                
                        obtiene=false
                    
                    end
                    i+=1
                end
            
                if(obtiene)
                
                    todos[actual].obtenerSalvoconducto(self)
                    salirDelMazo
                
                end
            end
            
        end

        def salirDelMazo
            
            @mazo.inhabilitarCartaEspecial(self)
              
        end
          
        def usada
          
            @mazo.habilitarCartaEspecial(self)
          
        end

        def toString

            return 'Sorpresa de salva de la carcel.'

        end


    end
end

File: CivitasJava/civitas/Sorpresa_SalirCarcel.java
package civitas;

import java.util.ArrayList;

class Sorpresa_SalirCarcel extends Sorpresa
{
    private MazoSorpresas mazo;

    Sorpresa_SalirCarcel(MazoSorpresas mazo)
    {
        super("Queda libre de la cárcel.");
        this.mazo=mazo;
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {
            informe(actual, todos);
            boolean obtiene=true;

            for(int i=0; i<todos.size()&&!obtiene; i++)
            {
                if(todos.get(i).tieneSalvoConducto())
                {
                    obtiene=false;
                }
            }

            if(obtiene)
            {
                todos.get(actual).obtenerSalvoConducto(this);
                salirDelMazo();
            }
        }
    }

    void salirDelMazo()
    {
        mazo.inhabilitarCartaEspecial(this);   
    }

    void usada()
    {
        mazo.habilitarCartaEspecial(this);   
    }

    @Override
    public String toString()
    {
        return "Sorpresa de salva de la cárcel.";
    }
}